active/TestInsecureHTTPVerbs.py

```diff
@@ -0,0 +1,71 @@
+activevulnerabilitysolution = 'Configure the web server to allow insecure methods like DELETE and PUT only for the relevant resources. ' + 'If your application does not need HTTP methods other than GET and POST, consider disabling the unused HTTP methods.'
+printdebugmessages = True
+#printdebugmessages= False
+
+def PrintAlerts(sas, msg, uri, insecureverb, responsestatuscode, responsestatusmessage):
+	attackevidence = 'VERB: \t' + insecureverb + '\t-- HTTP STATUS CODE: '+ str(responsestatuscode) +' -- \tMESSAGE: ' + responsestatusmessage  + '\t\t-- POTENTIALLY VULNERABLE: YES\n'
+	if printdebugmessages:
+		print (attackevidence)
+	sas.raiseAlert(1, 2, activevulnerabilitytitle, activevulnerabilityfulldescription, uri, 'HTTP VERB', 'ZAP sent an HTTP request with method - ' + insecureverb, '', activevulnerabilitysolution, attackevidence, 0, 0, msg);
+
+	
+def ProcessAndPrintAlert(sas, msg, uri, insecureverb):
+	if not msg.getResponseHeader().getStatusCode() in acceptedhttpstatuscodesforinsecureverbs:
+		PrintAlerts(sas, msg, uri, insecureverb, msg.getResponseHeader().getStatusCode(), msg.getResponseHeader().getReasonPhrase())
+
+
+def PrepareHttpRequest(msg, insecureverb):
+	msg.mutateHttpMethod(insecureverb)
+	if insecureverb == "POST" or insecureverb == "PUT":
+		msg.setRequestBody("bodytext")
+	if printdebugmessages:
+		print('After mutating resulting HTTP VERB: -- \t' + msg.getRequestHeader().getMethod() + '\n')
+
+	
+def TestTheURIForInsecureVerbs(sas, msg, uri, insecureverbs):
+	for insecureverb in insecureverbs:
+		try:
+			if printdebugmessages:
+				print ('Testing Initiated for the HTTP VERB: -- \t' + insecureverb + '\n')
+			PrepareHttpRequest(msg, insecureverb)
+			sas.sendAndReceive(msg)
+			if printdebugmessages:
+				print('VERB: \t' + insecureverb + '\t-- STATUS: '+ str(msg.getResponseHeader().getStatusCode()) +' -- \tMESSAGE: ' + msg.getResponseHeader().getReasonPhrase() + '\n')
+			ProcessAndPrintAlert(sas, msg, uri, insecureverb)
+		except Exception, e:
+			print('ERROR For: ' + insecureverb + 'Detail: ' + e.message + '\n')
+
+
+def scanNode(sas, msg):
+	url = msg.getRequestHeader().getURI().toString()
+	if printdebugmessages:
+    		print('Active scan script called for url = ' + url + '\n');
+		print('Testing for insecure verbs against a list of accepted http response status codes for the url = ' + url + '\n')
+    	msg = msg.cloneRequest();
```

Comment:
This should be done before `PrepareHttpRequest` in `TestTheURIForInsecureVerbs` instead of here, we need a new message for each test (otherwise the message shown in the active scan and alert might not be correct).